File: data/multilang_comparison/program_julia/src/func26.c
#include "program_julia.h" 
array_t* func26(array_t_param* vars, const unsigned long PATH0, int loopsFactor) {
   size_t pCounter = vars->size;
   if(PATH0 & 1) {
      array_t* array110;
      if (pCounter > 0) {
         array110 = vars->data[--pCounter];
         array110->refC++;
         DEBUG_COPY(array110->id);
      } else {
         array110 = (array_t*)malloc(sizeof(array_t));
         array110->size = 590;
         array110->refC = 1;
         array110->id = 110;
         array110->data = (unsigned int*)malloc(array110->size*sizeof(unsigned int));
         memset(array110->data, 0, array110->size*sizeof(unsigned int));
         DEBUG_NEW(array110->id);
      }
      unsigned int loop34 = 0;
      unsigned int loopLimit34 = (rand()%loopsFactor)/3 + 1;
      for(; loop34 < loopLimit34; loop34++) {
         for (int i = 0; i < array110->size; i++) {
            array110->data[i]--;
         }
         for (int i = 0; i < array110->size; i++) {
            if (array110->data[i] == 19) { 
               return array110;
            }
         }
      }
      array_t_param params0;
      params0.size = 1;
      params0.data = (array_t**)malloc(params0.size*sizeof(array_t*));
      params0.data[0] = array110;
      array_t* array111 = func36(&params0, get_path(), loopsFactor);
      DEBUG_RETURN(array111->id);
      free(params0.data);
      array111->refC--;
      if(array111->refC == 0) {
         free(array111->data);
         free(array111);
         DEBUG_FREE(array111->id);
      }
      array110->refC--;
      if(array110->refC == 0) {
         free(array110->data);
         free(array110);
         DEBUG_FREE(array110->id);
      }
   }
   else {
      array_t_param params0;
      params0.size = 0;
      params0.data = (array_t**)malloc(params0.size*sizeof(array_t*));
      array_t* array112 = func32(&params0, get_path(), loopsFactor);
      DEBUG_RETURN(array112->id);
      free(params0.data);
      array112->refC--;
      if(array112->refC == 0) {
         free(array112->data);
         free(array112);
         DEBUG_FREE(array112->id);
      }
   }
   array_t* array113;
   if (pCounter > 0) {
      array113 = vars->data[--pCounter];
      array113->refC++;
      DEBUG_COPY(array113->id);
   } else {
      array113 = (array_t*)malloc(sizeof(array_t));
      array113->size = 4;
      array113->refC = 1;
      array113->id = 113;
      array113->data = (unsigned int*)malloc(array113->size*sizeof(unsigned int));
      memset(array113->data, 0, array113->size*sizeof(unsigned int));
      DEBUG_NEW(array113->id);
   }
   return array113;
}

File: data/multilang_comparison/program_julia/src/func26.jl
include("program_julia_head.jl")
function func26(vars::JArray_param, PATH0::UInt64, loopsFactor::Int)::JArray
   pCounter = length(vars.data)
   if (PATH0 & 1) != 0
      array78 = nothing
      if pCounter > 1
          array78 = vars.data[pCounter]
          pCounter -= 1
          array78.refC += 1
      else
          array78 = JArray(zeros(Int, 924), 924, 1, 78)
      end
      loopLimit25 = (50)/3 + 1
      for i in 1:loopLimit25 
         for i in 0:(array78.size-1) 
            array78.data[i+1] -= 1
         end
         for i in 1:array78.size
             if array78.data[i] == 24
                     return array78
            end
         end
      end
      params0 = JArray_param{Int}()
      params0.size = 1
      push!(params0.data, array78)
      params0.size += 1
      array79 = func36(params0, get_path(), loopsFactor);
      array79.refC -= 1
      if array79.refC == 0
          empty!(array79.data)
          array79 = nothing
      end
      array78.refC -= 1
      if array78.refC == 0
          empty!(array78.data)
          array78 = nothing
      end
   
   else 
      params0 = JArray_param{Int}()
      params0.size = 0
      array80 = func32(params0, get_path(), loopsFactor);
      array80.refC -= 1
      if array80.refC == 0
          empty!(array80.data)
          array80 = nothing
      end
   end
   array81 = nothing
   if pCounter > 1
       array81 = vars.data[pCounter]
       pCounter -= 1
       array81.refC += 1
   else
       array81 = JArray(zeros(Int, 379), 379, 1, 81)
   end
   return array81;
end

